rotkehlchen/db/dbhandler.py

```diff
@@ -2768,27 +2768,49 @@ def get_latest_asset_value_distribution(self) -> List[DBAssetBalance]:
             ignored_assets = self.get_ignored_assets(write_cursor)
 
         with self.conn.read_ctx() as cursor:
+            treat_eth2_as_eth = self.get_settings(cursor).treat_eth2_as_eth
             cursor.execute(
                 'SELECT time, currency, amount, usd_value, category FROM timed_balances WHERE '
                 'time=(SELECT MAX(time) from timed_balances) AND category = ? ORDER BY '
                 'CAST(usd_value AS REAL) DESC;',
                 (BalanceType.ASSET.serialize_for_db(),),  # pylint: disable=no-member
             )
             asset_balances = []
+            eth_balance: Dict[str, Union[str, int]] = {
+                'time': 0,
+                'amount': '0',
+                'usd_value': '0',
+            }
             for result in cursor:
                 asset = Asset(result[1])
                 if asset in ignored_assets:
                     continue
+                # show eth & eth2 as eth in value distribution by asset
+                if treat_eth2_as_eth is True and asset in (A_ETH, A_ETH2):
+                    eth_balance['time'] = int(result[0])
+                    eth_balance['amount'] = str(FVal(eth_balance['amount']) + FVal(result[2]))
```

Comment:
Again here just like I asked you the last time you touched some code ilke this since now `DBAssetBalance` is being used in calculations it should have FVals and should be properly serialized and deserialized just like the change you did with `SingleDBAssetBalance`.

There is no point in doing what you are doing here ... you are turning string to FVal, add them and then back to strings. This means that the data structure needs to change.